src/Sylius/Behat/Context/Api/Shop/AddressContext.php

```diff
@@ -163,6 +172,116 @@ public function iSetTheAddressOfAsDefault(string $fullName): void
         $this->customerClient->update();
     }
 
+    /**
+     * @When /^I try to edit the (address of "([^"]+)")$/
+     */
+    public function iTryToEditTheAddressOf(AddressInterface $address)
+    {
+        $this->addressClient->buildUpdateRequest((string) $address->getId());
+    }
+
+    /**
+     * @When I change the first name to :firstName
+     */
+    public function iChangeTheFirstNameTo(string $firstName)
+    {
+        $this->addressClient->addRequestData('firstName', $firstName);
+    }
+
+    /**
+     * @When I change the last name to :lastName
+     */
+    public function iChangeTheLastNameTo(string $lastName)
+    {
+        $this->addressClient->addRequestData('lastName', $lastName);
+    }
+
+    /**
+     * @When I change the street to :street
+     */
+    public function iChangeTheStreetTo(string $street)
+    {
+        $this->addressClient->addRequestData('street', $street);
+    }
+
+    /**
+     * @When I change the city to :city
+     */
+    public function iChangeTheCityTo(string $city)
+    {
+        $this->addressClient->addRequestData('city', $city);
+    }
+
+    /**
+     * @When I change the postcode to :postcode
+     */
+    public function iChangeThePostcodeTo(string $postcode)
+    {
+        $this->addressClient->addRequestData('postcode', $postcode);
+    }
+
+    /**
+     * @When I choose :provinceName as my province
+     */
+    public function iChooseAsMyProvince(string $provinceName)
+    {
+        /** @var ProvinceInterface */
+        $province = $this->provinceProvider->findByName($provinceName);
+
+        $this->addressClient->addRequestData('provinceName', $province->getName());
+        $this->addressClient->addRequestData('provinceCode', $province->getCode());
+    }
+
+    /**
+     * @When I specify :provinceName as my province
+     */
+    public function iSpecifyProvince(string $provinceName)
+    {
+        $this->addressClient->addRequestData('provinceName', $provinceName);
+    }
+
+    /**
+     * @Then it should contain country :countryCode
+     */
+    public function itShouldContainCountry(string $countryCode)
+    {
+        Assert::true(
+            $this->containsValue(
+                $this->responseChecker->getResponseContent($this->addressClient->getLastResponse())['hydra:member'][0],
```

Comment:
As you are using responseChecker here I would suggest play a little with its method `getCollection` or `getCollectionItemsWithValue` they might be very nice in here 🚀

Reply:
its using getCollection now :D